tests/test_config_section.py

```diff
@@ -0,0 +1,399 @@
+# -*- coding: utf-8 -*-
+
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""
+Test validating of configuration values.
+"""
+
+from base import TestBase
+from mock import Mock
+from virtwho.config import ConfigSection, ValidationState
+
+MY_SECTION_NAME = 'my_section'
+
+MY_SECTION_VALUES = {
+    'my_str': 'foo',
+    'my_bool': False,
+    'my_list': ['cat', 'dog', 'frog'],
+    'must_have': 'bar'
+}
+
+
+class MyConfigSection(ConfigSection):
+    """
+    Example of ConfigSection subclass used for unit testing
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(MyConfigSection, self).__init__(*args, **kwargs)
+        self.add_key(key='my_str', validation_method=self._validate_non_empty_string,
+                     default='bar')
+        self.add_key(key='my_bool', validation_method=self._validate_str_to_bool, default=True)
+        self.add_key(key='must_have', validation_method=lambda *args: None,
+                     required=True)
+        self.add_key(key="my_list", validation_method=self._validate_list, default=[])
+
+
+class TestConfigSection(TestBase):
+    """
+    Class used for testing of ConfigSection
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(TestConfigSection, self).__init__(*args, **kwargs)
+        self.my_config = None
+
+    def init_config_section(self):
+        """
+        This method is executed before each unit test
```

Comment:
Is this comment true?  If so, we could just put this functionality in `setup`.  If not, we should remove the comment.